jdaviz/configs/specviz/plugins/viewers.py

```diff
@@ -458,9 +462,46 @@ def _plot_uncertainties(self):
                 # Add error lines to viewer
                 self.figure.marks = list(self.figure.marks) + [error_line_mark]
 
+    @property
+    def xunit(self):
+        return self._xunit if self._xunit is not None else u.Unit()
+
+    @property
+    def yunit(self):
+        return self._yunit if self._yunit is not None else u.Unit()
+
+    def set_display_units(self, xunit=None, yunit=None):
+        prev_xunit, prev_yunit = self.xunit, self.yunit
+        prev_xmin, prev_xmax = self.state.x_min, self.state.x_max
+        prev_ymin, prev_ymax = self.state.y_min, self.state.y_max
+
+        for mark in self.figure.marks:
+            if hasattr(mark, 'set_display_units'):
+                if mark._native_xunit is None:
+                    mark.set_native_units(xunit=self.xunit, yunit=self.yunit)
```

Comment:
this is a bit ugly... but since we don't have any hook here to the Line being initialized, we cannot set the initial units.  So instead, we just set them on the first iteration (and `viewer.xunit/yunit` are set in `set_plot_axes`.... also not an ideal location to handle that.